[BUG]: C++ examples fail to compile in release container on ARM

### Version

25.06

### Which installation method(s) does this occur on?

Docker

### Describe the bug.

Fails to link with `cannot find -lmorpheus` even though  `-Wl,-rpath,/opt/conda/envs/morpheus/lib` is specified.

Issue appears to be ARM specific, adding a call to `morpheus_utils_initialize_install_prefix()` in `examples/data_loss_prevention/CMakeLists.txt` doesn't resolve the issue.

### Minimum reproducible example

```shell
./docker/build_container_release.sh 
./docker/run_container_release.sh

# Within the container
conda env update --solver=libmamba \
  -n ${CONDA_DEFAULT_ENV} \
  --file ./conda/environments/all_cuda-128_arch-$(arch).yaml

./examples/data_loss_prevention/compile.sh
```

### Relevant log output

```
[0/2] Re-checking globbed directories...  
[9/11] Linking CXX shared module dlp_stages/_lib/regex_processor.cpython-312-aarch64-linux-gnu.so
FAILED: dlp_stages/_lib/regex_processor.cpython-312-aarch64-linux-gnu.so
: && /opt/conda/envs/morpheus/bin/c++ -fPIC   -flto -shared  -o dlp_stages/_lib/regex_processor.cpython-312-aarch64-linux-gnu.so dlp_stages/_lib/CMakeFiles/dlp_stages._lib.re
gex_processor.dir/regex_processor.cpp.o  -Wl,-rpath,/opt/conda/envs/morpheus/lib  -lmorpheus  /opt/conda/envs/morpheus/lib/libcudf.so  /opt/conda/envs/morpheus/lib/libglog.so.0.7.1  /opt/conda/envs/morpheus/lib/libcudart.so  -ldl  /opt/conda/envs/morpheus/lib/libgflags.so.2.2.2  -lpthread  -Wl,-rpath-link,/opt/conda/envs/morpheus/lib && cd /workspace/examples/data_loss_prevention/build/dlp_stages/_lib && /opt/conda/envs/morpheus/bin/strip /workspace/examples/data_loss_prevention/build/dlp_stages/_lib/regex_processor.cpython-312-aarch64-linux-gnu.so && cd /workspace/examples/data_loss_prevention/build/dlp_stages/_lib && /opt/conda/envs/morpheus/bin/cmake -E copy /workspace/examples/data_loss_prevention/build/dlp_stages/_lib/regex_processor.cpython-312-aarch64-linux-gnu.so /workspace/examples/data_loss_prevention/dlp_stages/_lib
/opt/conda/envs/morpheus/bin/../lib/gcc/aarch64-conda-linux-gnu/12.1.0/../../../../aarch64-conda-linux-gnu/bin/ld: cannot find -lmorpheus
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```


### Full env printout

<details><summary>Click here to see environment details</summary><pre>

 [Paste the results of print_env.sh here, it will be hidden by default]

</pre></details>


### Other/Misc.

_No response_

### Code of Conduct

- [x] I agree to follow Morpheus' Code of Conduct
- [x] I have searched the [open bugs](https://github.com/nv-morpheus/Morpheus/issues?q=is%3Aopen+is%3Aissue+label%3Abug) and have found no duplicates for this bug report

--- a/examples/data_loss_prevention/dlp_stages/_lib/CMakeLists.txt
+++ b/examples/data_loss_prevention/dlp_stages/_lib/CMakeLists.txt
@@ -15,7 +15,7 @@
 
 list(APPEND CMAKE_MESSAGE_CONTEXT "dlp_regex_processor")
 
-if(PROJECT_IS_TOPLEVEL)
+if(PROJECT_IS_TOP_LEVEL)
   find_library(MORPHEUS morpheus REQUIRED)
 else()
   set(MORPHEUS morpheus)

--- a/examples/developer_guide/3_simple_cpp_stage/src/simple_cpp_stage/_lib/CMakeLists.txt
+++ b/examples/developer_guide/3_simple_cpp_stage/src/simple_cpp_stage/_lib/CMakeLists.txt
@@ -15,7 +15,7 @@
 
 list(APPEND CMAKE_MESSAGE_CONTEXT "pass_thru_cpp_stage")
 
-if(PROJECT_IS_TOPLEVEL)
+if(PROJECT_IS_TOP_LEVEL)
   find_library(MORPHEUS morpheus REQUIRED)
 else()
   set(MORPHEUS morpheus)

--- a/examples/developer_guide/4_rabbitmq_cpp_stage/src/rabbitmq_cpp_stage/_lib/CMakeLists.txt
+++ b/examples/developer_guide/4_rabbitmq_cpp_stage/src/rabbitmq_cpp_stage/_lib/CMakeLists.txt
@@ -15,7 +15,7 @@
 
 list(APPEND CMAKE_MESSAGE_CONTEXT "rabbitmq_cpp_stage")
 
-if(PROJECT_IS_TOPLEVEL)
+if(PROJECT_IS_TOP_LEVEL)
   find_library(MORPHEUS morpheus REQUIRED)
 else()
   set(MORPHEUS morpheus)

